### PR TITLE
Adeleke/ltrgtr 167/UI updates

### PR DESCRIPTION
**Pull request by aabolade**
## Description

Implementing UI updates to match with suggested [Figma designs](https://www.figma.com/file/HP3BWmFoZykzyNld4Vqsuv/Notifications-for-IOS?node-id=2%3A1203&t=IIJeIdwyf1wEYEa1-1)

## Screenshots

Screenshot 2023-04-21 at 23 16 27
---

### Checklist

**Internationalization:** _(all strings must be localised. See [docs](https://github.com/noom/mobile-handbook/blob/master/general/tms.md))_
- [ ] All my strings were added to a `.strings` file so Smartling can initiate its workflow.

**Experimentation flags:** _List the ECT and/or Firebase flag names._
- [ ] I have created a RemoteConfig kill-switch if my experiment uses ECT.

**Unit tests:**
- [ ] My changes are covered with unit tests. If not, here's why: _Your explanation for why unit tests are unnecessary._

### CODE REVIEW

File: UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/Noom Tooltip/Extensions/Fonts.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    static func noomFontMedium(ofSize size: CGFloat) -> Self {
+        let uiFont = UIFont(name: "UntitledSans-Medium", size: size) ?? .systemFont(ofSize: size, weight: .medium)
+        return Font(uiFont as CTFont)
+    }
+}
+
+extension UIFont {
+    static let noomRegular = UIFont.systemFont(ofSize: 14)
+    static let noomRegularBold = UIFont.systemFont(ofSize: 14, weight: .bold)
```
Comment (ilyap7o) [inline UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/Noom Tooltip/Extensions/Fonts.swift:23]: Should this be using `noomFontRegular(...)` above?

If not, this naming is confusing as this is system font, not Noom

File: UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/NotificationRow.swift [hunk 2]
```diff
@@ -37,6 +41,27 @@ struct NotificationRow: ReducerProtocol {
                 state.notification.postId
                     .map { client.openNotification($0).fireAndForget() } ?? .none
             )
+        case .fetchImage:
+            guard let url = state.notification.imageUrl else { return .none }
+                        return URLSession.shared.dataTaskPublisher(for: url)
```
Comment (ilyap7o) [inline UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/NotificationRow.swift:46]: Fix intentation

File: UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/NotificationRow.swift [hunk 3]
```diff
@@ -37,6 +41,27 @@ struct NotificationRow: ReducerProtocol {
...
+                               return Action.imageFetched(dataImage)
+                           case .failure:
+                               return .imageFailed
+                           }
+                       })
+                       .eraseToEffect()
+        case .imageFetched(let image):
+            state.image = image
+            return .none
+        case .imageFailed: return .none
```
Comment (ilyap7o) [inline UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/NotificationRow.swift:64]: I don't think we need `case imageFailed` if we're not doing anything. Moot if you combine both to `.imageResponse(let result)`

File: UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/NotificationRow.swift [hunk 4]
```diff
@@ -37,6 +41,27 @@ struct NotificationRow: ReducerProtocol {
                 state.notification.postId
                     .map { client.openNotification($0).fireAndForget() } ?? .none
             )
+        case .fetchImage:
+            guard let url = state.notification.imageUrl else { return .none }
+                        return URLSession.shared.dataTaskPublisher(for: url)
+                       .catchToEffect()
+                       .receive(on: RunLoop.main)
```
Comment (ilyap7o) [inline UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/NotificationRow.swift:48]: Add `@Dependency(\.mainQueue) var mainQueue` in the reducers `var`s, and do `.receive(on: mainQueue` here

File: UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/NotificationRow.swift [hunk 5]
```diff
@@ -26,6 +27,9 @@ struct NotificationRow: ReducerProtocol {
 
     enum Action {
         case rowTapped
+        case fetchImage
+        case imageFailed
+        case imageFetched(UIImage)
```
Comment (ilyap7o) [inline UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/NotificationRow.swift:32]: Combine these into
```
case imageResponse(Result<Data, Error>)
```

Then below, you can simply do:
```
return URLSession.shared.dataTaskPublisher(for: url)
    .recieve(on: mainQueue)
    .catchToEffect(Action.imageResponse)
```
(Global search our codebase for `.catchToEffect(Action.` for examples)

Then, handle the `switch result` in `case .imageResponse`, setting the state image if there's a valid one

**Review by ilyap7o — CHANGES_REQUESTED** [review]

File: UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/Noom Tooltip/Extensions/Fonts.swift (hunk 1 above)
Comment (aabolade) [reply UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/Noom Tooltip/Extensions/Fonts.swift:23]: Fair enough, I think this extension may be redundant tbh

File: UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/NotificationRow.swift [hunk 6]
```diff
@@ -61,31 +87,87 @@ extension NotificationRow {
...
-                        Text(DateFormatter.yyyymmdd.string(from: viewStore.lastUpdate))
-                    }
-                    Spacer()
+                VStack(spacing: 12) {
+                    HStack(spacing: 8) {
+                        if let image = viewStore.image {
+                            Image(uiImage: image)
+                                .frame(width: 28, height: 28)
+                        } else {
+                            Circle().fill(Color(.backgroundGrey))
```
Comment (ilyap7o) [inline UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/NotificationRow.swift:127]: (Nit) `.fill(...)` on new line

File: UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/Noom Tooltip/Extensions/Colors.swift [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+// Created by Leke Abolade
+// Copyright ® 2023 Noom, Inc. All Rights Reserved
+//
+
+import SwiftUI
+
+extension UIColor {
+    static let tarocco = UIColor(red: 0.984, green: 0.318, blue: 0.231, alpha: 1)
+    static let backgroundGrey = UIColor(red: 0.851, green: 0.851, blue: 0.851, alpha: 1)
+    static let greyThree = UIColor(red: 0.4, green: 0.4, blue: 0.4, alpha: 1)
```
Comment (ilyap7o) [inline UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/Noom Tooltip/Extensions/Colors.swift:11]: (Nit) Call it `grey3` to be consistent with `coach-ios`

File: UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/NotificationRow.swift [hunk 8]
```diff
@@ -37,6 +42,21 @@ struct NotificationRow: ReducerProtocol {
                 state.notification.postId
                     .map { client.openNotification($0).fireAndForget() } ?? .none
             )
+        case .fetchImage:
+            guard let url = state.notification.imageUrl else { return .none }
+            return URLSession.shared.dataTaskPublisher(for: url)
+                .receive(on: mainQueue)
+                .map(\.data)
+                .mapError { $0 }
```
Comment (ilyap7o) [inline UpstraUIKit/UpstraUIKit/Modules/Comunity/Community Notifications/NotificationRow.swift:50]: What does this do? I thought `catchToEffect` would type-erase everything as needed 🤷‍♂️

**Review by ilyap7o — APPROVED** [review]
LG! Nits only

**Pull request merged**